gluten-core/src/main/scala/org/apache/gluten/execution/SortUtils.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gluten.execution
+
+import org.apache.spark.sql.execution.{ProjectExec, SortExec, SparkPlan}
+
+object SortUtils {
+  def dropPartialSort(plan: SparkPlan): SparkPlan = plan match {
+    case PartialSortChild(p) => p
+    case ProjectChild(PartialSortChild(ProjectChild(p))) if plan.outputSet == p.outputSet => p
```

Comment:
nit:

```
case PartialSortLike(child) => child
case ProjectLike(PartialSortLike(ProjectLike(child))) if plan.outputSet == child.outputSet => child
```

For code readability. Also, let's add some comments to demostrate where the projects came from (e.g., from pre/post project-pulling)